### PR TITLE
Mappings for 1.18 mountain surface builders

### PR DESCRIPTION
**Pull request by mschae23**
This PR adds mappings for 1.18 mountain surface builders.

It renames `NewMountainSurfaceBuilder` to `AbstractMountainSurfaceBuilder`.

### CODE REVIEW

**Comment by liach** [comment]
Should match the id of the surface builders than biomes, as they are surface builders than biomes. Can the surface builders by used by any biome? If yes, then the first two renames has even less reason.

**Comment by mschae23** [comment]
> Can the surface builders by used by any biome? If yes, then the first two renames has even less reason.

They can. I thought that this change could be good for the same reason that the biomes were renamed, to reduce confusion between old mountains and new mountains. But if the name should be like the registry ID, I can revert the renames.

**Comment by liach** [comment]
Hmm, matching registry ID isn't perfect too if the another instance of the class can be assigned to another registry ID. The best way is still to approach by the builder's behavior and be descriptive; if it makes the terrain "extreme" then the renames are perfectly fine.

**Comment by mschae23** [comment]
> The best way is still to approach by the builder's behavior and be descriptive; if it makes the terrain "extreme" then the renames are perfectly fine.

The surface builder changes the blocks at the surface. The `mountain` one makes it have sometimes grass, sometimes stone. I don't know how I would describe that in the name.

**Comment by liach** [comment]
Then I think keeping the old `MountainSurfaceBuilder` name should be fine, as we can use the registry id as a second-hand evidence.

**Comment by mschae23** [comment]
> Then I think keeping the old `MountainSurfaceBuilder` name should be fine, as we can use the registry id as a second-hand evidence.

Sure! I initially renamed them because I wanted to also rename `NewMountainSurfaceBuilder` to `MountainSurfaceBuilder`, but `AbstractMountainSurfaceBuilder` is better for that class anyway.

**Comment by liach** [comment]
Hmm, does this "update base" functionality work for experimental versions, I wonder?... Rip, it doesn't

**Comment by mschae23** [comment]
> Hmm, does this "update base" functionality work for experimental versions, I wonder?... Rip, it doesn't

I fixed it now.

**Comment by liach** [comment]
Mind change branch to snapshot 4?

**Comment by mschae23** [comment]
> Mind change branch to snapshot 4?

Done.

File: mappings/net/minecraft/world/gen/surfacebuilder/StoneShoreSurfaceBuilder.mapping [hunk 1]
```diff
@@ -0,0 +1,8 @@
+CLASS net/minecraft/class_6485 net/minecraft/world/gen/surfacebuilder/StoneShoreSurfaceBuilder
```
Comment (apple502j) [inline mappings/net/minecraft/world/gen/surfacebuilder/StoneShoreSurfaceBuilder.mapping]: This is also used by stony peaks (as parent) iirc - I mapped it `StoneSurfaceBuilder`.

**Comment by mschae23** [comment]
Updated to experimental snapshot 5.

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**